src/consts/Stages.ts

```diff
@@ -184,57 +193,159 @@ export const STAGES: ReadonlyArray<STAGE> = [
+          const next = thisStage?.subStages
+            ?.find(({ key }) => key === "SelectionPackageHQApproval")
+            ?.next?.(request);
+          const nextPrettyName = thisStage?.subStages?.find(
+            ({ key }) => key === next
+          )?.text;
+          return `Forward Stage Change: HQ Approval to ${nextPrettyName}`;
+        },
+        readyForNext: () => true,
+        previous: "DraftPackageHRL",
+        previousEventTitle:
+          "Backward Stage Change: HQ Approval to Draft Package (HRL)",
+        showStage: (request) => request?.hqApproval === "Yes",
+      },
+      {
+        key: "SelectionPackageCAApproval",
+        text: "CA Approval",
+        next: (request) => {
           if (request?.titleV === "Yes") {
-            return "Forward Stage Change: Package Approval to Title V";
+            return "TitleV";
+          } else {
+            return STAGES.find(({ key }) => key === "PackageApproval")
+              ?.subStages?.find(({ key }) => key === "TitleV")
+              ?.next?.(request);
           }
-          return undefined;
+        },
+        nextEventTitle: (request) => {
+          const thisStage = STAGES.find(({ key }) => key === "PackageApproval");
+          const next = thisStage?.subStages
+            ?.find(({ key }) => key === "SelectionPackageCAApproval")
+            ?.next?.(request);
+          const nextPrettyName = thisStage?.subStages?.find(
+            ({ key }) => key === next
+          )?.text;
+          return `Forward Stage Change: CA Approval to ${nextPrettyName}`;
         },
         readyForNext: () => true,
-        previous: "SelectionPackageOSFApproval",
+        previous: "DraftPackageHRL",
         previousEventTitle:
-          "Backward Stage Change: Package Approval to OSF Approval",
+          "Backward Stage Change: CA Approval to Draft Package (HRL)",
+        showStage: (request) => request?.csfcaApproval === "Yes",
       },
+
       {
         key: "TitleV",
         text: "Title V",
         next: undefined,
         nextEventTitle: undefined,
         readyForNext: () => true,
-        previous: "PackageApproval",
+        previous: "DraftPackageHRL",
         previousEventTitle:
-          "Backward Stage Change: Title V to Package Approval",
+          "Backward Stage Change: Title V to Draft Package (HRL)",
+        showStage: (request) => request?.titleV === "Yes",
       },
     ],
+    showStage: () => true,
   },
   {
     key: "Complete",
     text: "Complete",
     next: undefined,
     nextEventTitle: undefined,
-    readyForNext: () => true,
+    readyForNext: () => false,
+    previous: undefined,
+    previousEventTitle: undefined,
+    showStage: () => true,
+  },
+  {
+    key: "Cancelled",
+    text: "Cancelled",
+    next: undefined,
+    nextEventTitle: undefined,
+    readyForNext: () => false,
     previous: undefined,
     previousEventTitle: undefined,
+    showStage: () => false,
```

Comment:
Need to implement the showStage check for the primary stages -- as right now it shows "Cancelled" on all requests because the StatusBar doesn't filter the primary stages, just the subStages.